test/Microsoft.SqlTools.ServiceLayer.IntegrationTests/ObjectManagement/DatabaseHandlerTests.cs

```diff
@@ -374,6 +378,90 @@ public async Task VerifyDatabaseScopedConfigurationsTest()
+                    var testDatabaseFiles = ObjectManagementTestUtils.GetTestDatabaseFiles();
+                    databaseFile.AddRange(testDatabaseFiles);
+
+                    // Attaching the files to the testdatabase
+                    testDatabase.Files = databaseFile.ToArray();
+
+                    // Validate the result files
+                    await ObjectManagementTestUtils.SaveObject(parametersForUpdate, testDatabase);
+                    DatabaseViewInfo updatedDatabaseViewInfo = await ObjectManagementTestUtils.GetDatabaseObject(parametersForUpdate, testDatabase);
+
+                    // verify the modified properties
+                    Assert.That(((DatabaseInfo)updatedDatabaseViewInfo.ObjectInfo).Files.Count, Is.EqualTo(4), $"Two files should exists, as we modified one and added two files");
+                    Assert.That(((DatabaseInfo)updatedDatabaseViewInfo.ObjectInfo).Files[0].SizeInMb, Is.EqualTo(15), $"Files updated value should match");
+                    Assert.That(((DatabaseInfo)updatedDatabaseViewInfo.ObjectInfo).Files[0].AutoFileGrowth, Is.EqualTo(74), $"Files updated value should match");
+
+                    Assert.That(((DatabaseInfo)updatedDatabaseViewInfo.ObjectInfo).Files[1].Id, Is.Not.EqualTo(0), $"Newly created file should have an Id");
+                    Assert.That(((DatabaseInfo)updatedDatabaseViewInfo.ObjectInfo).Files[1].Name, Is.EqualTo(testDatabaseFiles[1].Name), $"Files updated value should match");
+
+                    Assert.That(((DatabaseInfo)updatedDatabaseViewInfo.ObjectInfo).Files[3].Id, Is.Not.EqualTo(0), $"Newly created file should have an Id");
+                    Assert.That(((DatabaseInfo)updatedDatabaseViewInfo.ObjectInfo).Files[3].Name, Is.EqualTo(testDatabaseFiles[0].Name), $"Files updated value should match");
+
+                    // Deleting newly created file
+                    databaseFile.RemoveAt(3);
+                    testDatabase.Files = databaseFile.ToArray();
+                    
+                    // Validate the result files
+                    await ObjectManagementTestUtils.SaveObject(parametersForUpdate, testDatabase);
+                    DatabaseViewInfo databaseViewInfoAfterFileDelete = await ObjectManagementTestUtils.GetDatabaseObject(parametersForUpdate, testDatabase);
+                    Assert.That(((DatabaseInfo)databaseViewInfoAfterFileDelete.ObjectInfo).Files.Count, Is.EqualTo(3), $"Should have only 3 files as we removed one");
+
+                    // cleanup
+                    await ObjectManagementTestUtils.DropObject(connectionResult.ConnectionInfo.OwnerUri, objUrn, throwIfNotExist: true);
+                    Assert.That(DatabaseExists(testDatabase.Name!, server), Is.False, $"Database '{testDatabase.Name}' was not dropped succesfully");
+                }
+                finally
+                {
+                    // Cleanup using SMO if Drop didn't work
+                    DropDatabase(server, testDatabase.Name!);
```

Comment:
It seems redundant to have two cleanups - we should just have one and then fail the test if it can't drop it.

Reply:
Is it ok to leave the database on server during the test failure? It is file till the STS tests pipelines are not scheduled daily.

Reply:
No, which is why I would say this is the one that should stay. 

Although for tests like this where we aren't testing the actual creating of the database itself have you considered using C:\src\sqltoolsservice\test\Microsoft.SqlTools.ServiceLayer.Test.Common\SqlTestDb.cs ? That will auto-drop once it's disposed which can make things simpler.

Reply:
Verified the SqlTestDb class and using it to create the database for files and dsc tests. Thanks